lib/install/helpers.rb

```diff
@@ -1,48 +1,106 @@
+# frozen_string_literal: true
+
 require 'json'
 
 module Helpers
+  TOOLS_COMMANDS = {
+    yarn: { cmd: 'yarn', run: 'yarn', x: 'npx' },
+    bun: { cmd: 'bun', run: 'bun run', x: 'bunx' }
+    pnpm: { cmd: 'pnpm', run: 'pnpm run', x: 'pnpx' },
+    npm: { cmd: 'npm', run: 'npm run', x: 'npx' },
+  }.freeze
+
+  SUPPORTED_TOOLS = TOOLS_COMMANDS.keys.freeze
+  DEFAULT_TOOL = :yarn
+
+  SCRIPT_STRATEGIES = {
+    yarn: lambda do |helper, name, script, run_script|
+      npx_version = `npx -v`.to_f
+
+      if npx_version >= 7.1 && npx_version < 8.0
+        helper.say "Add #{name} script"
+        helper.run %(npm set-script #{name} "#{script}")
+        helper.run %(#{helper.bundler_run_cmd} #{name}) if run_script
+      elsif npx_version >= 8.0
+        helper.say "Add #{name} script"
+        helper.run %(npm pkg set scripts.#{name}="#{script}")
+        helper.run %(#{helper.bundler_run_cmd} #{name}) if run_script
+      else
+        helper.say %(Add "scripts": { "#{name}": "#{script}" } to your package.json), :green
+      end
+    end,
+
+    bun: lambda do |helper, name, script, run_script|
+      package_json = JSON.parse(File.read("package.json"))
+      package_json["scripts"] ||= {}
+      package_json["scripts"][name] = script.gsub('\\"', '"')
+      File.write("package.json", JSON.pretty_generate(package_json))
+      helper.run %(#{helper.bundler_run_cmd} #{name}) if run_script
+    end,
+
+    pnpm: lambda do |helper, name, script, run_script|
+      helper.say "Add #{name} script"
+      helper.run %(pnpm pkg set scripts.#{name}="#{script}")
+      helper.run %(#{helper.bundler_run_cmd} #{name}) if run_script
+    end,
+
+    npm: lambda do |helper, name, script, run_script|
+      helper.say "Add #{name} script"
+      npx_version = `npx -v`.to_f
+
+      if npx_version >= 7.1 && npx_version < 8.0
+        helper.run %(npm set-script #{name} "#{script}")
+      else
+        helper.run %(npm pkg set scripts.#{name}="#{script}")
+      end
+
+      helper.run %(#{helper.bundler_run_cmd} #{name}) if run_script
+    end
+  }.freeze
```

Comment:
This is interesting, but too complicated.

Reply:
I get what you mean, i commited a new version a bit more simpler to read.